File: p015.py
#!/usr/bin/python


def route_num(cube_size):
    l = [1] * cube_size
    for i in range(cube_size):
        for j in range(i):
            l[j] = l[j] + l[j - 1]
        l[i] = 2 * l[i - 1]
        print(l[i])
    return l[cube_size - 1]


print(route_num(20))

File: rust/src/bin/p015.rs

fn route_num(cube_size: usize) -> u64 {
    let mut l: Vec<u64> = vec![1; cube_size];
    for i in (0..cube_size) {
        for j in (0..i) {
            if j > 0 {
                l[j] = l[j] + l[j - 1];
            } else {
                l[j] = l[j] + 1;
            }
            // println!("j - {}", l[j]);
        }
        if i > 0 {
            l[i] = 2 * l[i - 1];
        } else {
            l[i] = 2;
        }
        // println!("i - {}", l[i]);
    }
    l[cube_size - 1]
}

fn main() {
    println!("{}", route_num(20));
}
